spring-xd-spark-streaming/src/main/java/org/springframework/xd/spark/streaming/examples/SparkLogModuleOptionsMetadata.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.spark.streaming.examples;
+
+import org.springframework.xd.module.options.spi.ModuleOption;
+import org.springframework.xd.spark.streaming.DefaultSparkStreamingModuleOptionsMetadata;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+public class SparkLogModuleOptionsMetadata extends DefaultSparkStreamingModuleOptionsMetadata {
```

Comment:
I probably don't fully understand what you have in mind by a Mixin alternative to this, but I can explain the idea. Any Spark Streaming module should have the DefaultSparkStreamingModuleOptionsMetadata since that provides common general settings like "batchInterval" as well as the indicator that it's actually a Spark Streaming module. However, if a developer is creating a custom implementation of Processor and that will accept some option values (like filePath here), they need the simplest possible way to add those. If they can do something that does not require an explicit extends, that would be even better, but can we then handle the mixin of the defaults within the framework so that is also hidden from the user?